test/test_transforms_task.py

```diff
@@ -713,3 +714,83 @@ def test_treeherder_defaults(run_transform, graph_config, kind, task_def, expect
     pprint(task_dict, indent=2)
 
     assert task_dict["task"].get("extra", {}).get("treeherder", {}) == expected_th
+
+
+@pytest.mark.parametrize(
+    "test_task, expectation",
+    (
+        (
+            {
+                "label": "task1",
+                "dependencies": ["dependency"] * 2,
+                "soft-dependencies": ["dependency"] * 1,
+                "if-dependencies": ["dependency"] * 4,
+            },
+            does_not_raise(),
+        ),
+        (
+            {
+                "label": "task2",
+                "dependencies": ["dependency"] * 97,
+                "soft-dependencies": ["dependency"] * 1,
+                "if-dependencies": ["dependency"] * 1,
+            },
+            does_not_raise(),
+        ),
+        (
+            {
+                "label": "task3",
+                "dependencies": ["dependency"] * 98,
+                "soft-dependencies": ["dependency"] * 1,
+                "if-dependencies": ["dependency"] * 1,
+            },
+            pytest.raises(Exception),
+        ),
+        (
+            {
+                "label": "task3",
+                "dependencies": ["dependency"] * 99,
+                "soft-dependencies": ["dependency"],
+                "if-dependencies": ["dependency"],
+            },
+            pytest.raises(Exception),
+        ),
+    ),
+)
+def test_check_task_dependencies(graph_config, test_task, expectation):
+    params = FakeParameters(
+        {
+            "base_repository": "git@github.com://github.com/mozilla/example.git",
+            "build_date": 0,
+            "build_number": 1,
+            "head_repository": "git@github.com://github.com/mozilla/example.git",
+            "head_rev": "abcdef",
+            "head_ref": "default",
+            "level": "1",
+            "moz_build_date": 0,
+            "next_version": "1.0.1",
+            "owner": "some-owner",
+            "project": "some-project",
+            "pushlog_id": 1,
+            "repository_type": "git",
+            "target_tasks_method": "test_method",
+            "tasks_for": "github-pull-request",
+            "try_mode": None,
+            "version": "1.0.0",
+        },
+    )
+
+    transform_config = TransformConfig(
+        "check_task_dependencies",
+        str(here),
+        {},
+        params,
+        {},
+        graph_config,
+        write_artifacts=False,
+    )
+
+    with expectation:
+        assert (
+            len(list(task.check_task_dependencies(transform_config, [test_task]))) == 1
+        )
```

Comment:
Quick question: What does this assertion verify?

Reply:
The "test_check_task_dependencies" is called for every 'test_task' and 'expectation' pair that we have defined using "@pytest.mark.parametrize" . For each call only 1 task is passed to "check_task_dependencies" and "check_task_dependencies" yields the tasks passed to it. Since only 1 task is being passed everytime during the test, the number of tasks returned should be 1 for each call of "test_check_task_dependencies".  Hence i am asserting that the length of list returned by "check_task_dependencies" should be 1. If there is an exception, it will be caught by the 'expectation'. Hope its clear!

Let me know if more clarification is needed!

Reply:
Thanks for the details. It sounds good to me 👍